### PR TITLE
Missing translation and localization update

### PR DESCRIPTION
**Pull request by Niracash**
Updated localization translation script to automate whole website to translate without problem.

Updates:
- Fixed a bug where FAQ arrays were not translating
- Added new parts name in english
- Updated script to translate whole website (Homepage, parts, FAQ) with command `php artisan localization:translate`

### CODE REVIEW

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 1]
```diff
@@ -8,36 +8,21 @@
...
-     * Execute the console command.
-     *
-     * @return int
-     */
     public function handle()
     {
-        $sourceLang = 'en'; // Source language code
-        $deeplApiKey = env('DEEPL_TRANSLATE'); // Fetching the DeepL API key from .env
+        $sourceLang = 'en';
+        $deeplApiKey = env('DEEPL_TRANSLATE');
```
Comment (nowalor) [inline app/Console/Commands/TranslateLocalizationFiles.php:20]: Please use a config file and don't read variables directly from env

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 2]
```diff
@@ -8,36 +8,21 @@
...
 
         if (!$deeplApiKey) {
-            $this->error('DeepL API key not found in .env file. Please set the DEEPL_TRANSLATE variable.');
+            $this->error('DeepL API key is missing in .env file.');
             return Command::FAILURE;
         }
 
-        // Define the target languages and their corresponding DeepL language codes
         $targetLangs = [
```
Comment (nowalor) [inline app/Console/Commands/TranslateLocalizationFiles.php:26]: Maybe consider using an enum for this, much cleaner imo

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 3]
```diff
@@ -47,93 +32,157 @@ public function handle()
             'pl' => 'PL', // Polish
         ];
 
-        // Path to the source file
-        $sourceFilePath = base_path("lang/{$sourceLang}.json");
+        $client = new Client(['base_uri' => 'https://api-free.deepl.com/v2/']);
```
Comment (nowalor) [inline app/Console/Commands/TranslateLocalizationFiles.php:35]: Please read the api url from and env variable and a config file. Also consider creating the client in the constructor

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 4]
```diff
@@ -47,93 +32,157 @@ public function handle()
...
-        $sourceContent = json_decode(File::get($sourceFilePath), true);
+        $this->info("All files have been translated successfully!");
+        return Command::SUCCESS;
+    }
 
-        if (empty($sourceContent)) {
-            $this->error("Source file {$sourceLang}.json is empty or invalid.");
-            return Command::FAILURE;
-        }
+    protected function translateJsonFile($sourceLang, $targetLangs, $client, $apiKey)
```
Comment (nowalor) [inline app/Console/Commands/TranslateLocalizationFiles.php:50]: much nicer to type hint all arguments and return types in modern PHP

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 5]
```diff
@@ -47,93 +32,157 @@ public function handle()
...
+    protected function translateFile($sourceLang, $file, $targetLangs, $client, $apiKey)
+    {
+        $sourcePath = base_path("lang/{$sourceLang}/{$file}");
+
+        if (!File::exists($sourcePath)) {
+            $this->error("Source file not found: {$sourcePath}");
+            return;
+        }
+
+        $sourceArray = require $sourcePath;
```
Comment (nowalor) [inline app/Console/Commands/TranslateLocalizationFiles.php:96]: Not sure about require it's very old school PHP, can you use the file helper or something to read the file?

**Review by nowalor — CHANGES_REQUESTED** [review]
Good job just coming up with a working solution and doing it so fast, of course since it's a review I mostly focus on the stuff that can be better but it's awesome what you have done. 

It's a bit hard reviewing it now since I didn't get a chance to review the initial feature. Here is one thought I had.

Instead of making the script dynamic for both PHP arrays and JSON couldn't it have been better to just decide on what structure fits better and stick with that?

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 6]
```diff
@@ -8,36 +8,21 @@
...
-    /**
-     * Execute the console command.
-     *
-     * @return int
-     */
     public function handle()
     {
-        $sourceLang = 'en'; // Source language code
-        $deeplApiKey = env('DEEPL_TRANSLATE'); // Fetching the DeepL API key from .env
+        $sourceLang = 'en';
```
[the remark this answers is not in the document]
Comment (Niracash) [reply app/Console/Commands/TranslateLocalizationFiles.php:18]: I put it in config file if that is ok?

[the remark this answers is not in the document]
Comment (nowalor) [reply app/Console/Commands/TranslateLocalizationFiles.php:18]: don't think it needs to be in the config

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 7]
```diff
@@ -8,132 +8,195 @@
...
-            'fr' => 'FR', // French
-            'se' => 'SV', // Swedish
-            'ge' => 'DE', // German
-            'it' => 'IT', // Italian
-            'pl' => 'PL', // Polish
-        ];
-
-        // Path to the source file
-        $sourceFilePath = base_path("lang/{$sourceLang}.json");
+        $this->sourceLang = config('deepl.source_language', 'en');
```
Comment (nowalor) [inline app/Console/Commands/TranslateLocalizationFiles.php]: Would make more sense to do in the constructor

File: app/Console/Commands/TranslateLocalizationFiles.php [hunk 8]
```diff
@@ -8,132 +8,195 @@
...
+        $this->client = new Client([
+            'base_uri' => config('deepl.base_url'),
+        ]);
+    }
 
-        if (!$deeplApiKey) {
-            $this->error('DeepL API key not found in .env file. Please set the DEEPL_TRANSLATE variable.');
+    public function handle(): int
+    {
+        $apiKey = config('deepl.key');
```
Comment (nowalor) [inline app/Console/Commands/TranslateLocalizationFiles.php:32]: should be a class variable and set in the constructor

File: config/deepl.php [hunk 9]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+return [
+
+    'key' => env('DEEPL_API_KEY', null),
+
+    'base_url' => env('DEEPL_API_BASE_URL', 'https://api-free.deepl.com/v2/'),
+
+    'source_language' => env('DEEPL_SOURCE_LANG', 'en'),
+    'target_languages' => [
```
Comment (nowalor) [inline config/deepl.php]: not normally what you would use a config for, it's only for api urls and keys and such. I get that you use these in the request but if you think about it they are also the website languages and we might want to use them in various places not just when dealing with the deepL API.

So WebsiteLocaleEnum.php would feel more fitting. I also see you are already refactoring to en Enum based on our slack convo but still seems like the comments could be helpful :)

File: config/deepl.php [hunk 10]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+return [
+
+    'key' => env('DEEPL_API_KEY', null),
+
+    'base_url' => env('DEEPL_API_BASE_URL', 'https://api-free.deepl.com/v2/'),
+
+    'source_language' => env('DEEPL_SOURCE_LANG', 'en'),
```
Comment (nowalor) [inline config/deepl.php]: Does not need to be a config variable imo, after you have switched to an enum for the localization variables just set source lang in the command to be able to LocaleEnum::EN->value or something like that

File: config/deepl.php [hunk 11]
```diff
@@ -0,0 +1,9 @@
+<?php
```
Comment (nowalor) [inline config/deepl.php:1]: I normally just put these in services.php, no need to create a new class

File: app/Enums/TargetLanguages.php [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+enum TargetLanguages: string
+{
+    case DK = 'DA'; // Danish
+    case FR = 'FR'; // French
+    case SE = 'SV'; // Swedish
+    case GE = 'DE'; // German
+    case IT = 'IT'; // Italian
+    case PL = 'PL'; // Polish
+
+    public static function getLanguages(): array
```
Comment (nowalor) [inline app/Enums/TargetLanguages.php:14]: Alternative that I have used to do a similar thing before,
 ```
   public static function values(): array
    {
        return array_map(fn ($case) => $case->value, self::cases());
    }
```

**Review by nowalor — APPROVED** [review]
Much fewer and more minor comments that last time, it's looking pretty good. If you fix everything you can merge and deploy it yourself :)

**Pull request merged**